### ISSUE TITLE
Support custom date format function for the input element

### ISSUE THREAD

**Issue by neelav** [primefaces/primereact#852]
### There is no guarantee in receiving an immediate response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeReact PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[ ] bug report
[ X] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

I have date formats that are using the momentjs format (which is slightly different than what primereact allows).  Would it be possible to either support momentjs formats or supply some sort of `formatDate` option where I can specify a function that can convert a javascript Date object to a string? This would be used to format the text displayed in the text input element.

**Comment by cagataycivici** [comment]
If we add a formatDate function, there will be a parseDate function as well so format and parse are handled on your side. We'll wait for more feedback to see if more users need this.

**Comment by melloware** [comment]
Implemented in 8.6.2: https://github.com/primefaces/primereact/issues/3381
